stubs/psycopg2/psycopg2/extras.pyi

```diff
@@ -58,31 +79,67 @@ class DictRow(list[Any]):
     def __reduce__(self): ...
 
 class RealDictConnection(_connection):
-    def cursor(self, *args, **kwargs): ...
+    @overload
+    def cursor(
+        self, name: str | bytes | None = ..., *, withhold: bool = ..., scrollable: bool | None = ...
+    ) -> RealDictCursor: ...
+    @overload
+    def cursor(
+        self,
+        name: str | bytes | None = ...,
+        *,
+        cursor_factory: Callable[..., _T_cur],
+        withhold: bool = ...,
+        scrollable: bool | None = ...,
+    ) -> _T_cur: ...
+    @overload
+    def cursor(
+        self, name: str | bytes | None, cursor_factory: Callable[..., _T_cur], withhold: bool = ..., scrollable: bool | None = ...
+    ) -> _T_cur: ...
 
 class RealDictCursor(DictCursorBase):
     def __init__(self, *args, **kwargs) -> None: ...
     column_mapping: Any
     def execute(self, query, vars: Any | None = ...): ...
     def callproc(self, procname, vars: Any | None = ...): ...
+    def fetchone(self) -> RealDictRow | None: ...  # type: ignore[override]
+    def fetchmany(self, size: int | None = ...) -> list[RealDictRow]: ...  # type: ignore[override]
+    def fetchall(self) -> list[RealDictRow]: ...  # type: ignore[override]
+    def __next__(self) -> RealDictRow: ...  # type: ignore[override]
 
 class RealDictRow(OrderedDict[Any, Any]):
     def __init__(self, *args, **kwargs) -> None: ...
     def __setitem__(self, key, value) -> None: ...
 
 class NamedTupleConnection(_connection):
-    def cursor(self, *args, **kwargs): ...
+    @overload
+    def cursor(
+        self, name: str | bytes | None = ..., *, withhold: bool = ..., scrollable: bool | None = ...
+    ) -> NamedTupleCursor: ...
+    @overload
+    def cursor(
+        self,
+        name: str | bytes | None = ...,
+        *,
+        cursor_factory: Callable[..., _T_cur],
+        withhold: bool = ...,
+        scrollable: bool | None = ...,
+    ) -> _T_cur: ...
+    @overload
+    def cursor(
+        self, name: str | bytes | None, cursor_factory: Callable[..., _T_cur], withhold: bool = ..., scrollable: bool | None = ...
+    ) -> _T_cur: ...
 
 class NamedTupleCursor(_cursor):
     Record: Any
     MAX_CACHE: int
     def execute(self, query, vars: Any | None = ...): ...
     def executemany(self, query, vars): ...
     def callproc(self, procname, vars: Any | None = ...): ...
-    def fetchone(self) -> tuple[Any, ...] | None: ...
-    def fetchmany(self, size: int | None = ...) -> list[tuple[Any, ...]]: ...
-    def fetchall(self) -> list[tuple[Any, ...]]: ...
-    def __iter__(self): ...
+    def fetchone(self) -> NamedTuple | None: ...  # type: ignore[override]
```

Comment:
Returning `typing.NamedTuple` is a little weird, but it apparently works with both pyright and mypy, so I suppose it's fine.

Reply:
Yes, I agree. But at least it offers a hint to the caller that they can use either list indexing or access the fields with dot-notation.